docs/secure-communication-agents.asciidoc

```diff
@@ -1,21 +1,353 @@
 [[secure-communication-agents]]
 == Secure communication with APM Agents
 
-To secure the communication between APM Agents and the APM Server:
+There are two ways to secure the communication between APM Agents and the APM Server.
+Both options can be enabled in parallel,
+allowing Elastic APM agents to chose whichever mechanism they support.
 
-. <<secret-token,Set a secret token in your Agents and Server>>
-. <<ssl-setup,Enable SSL/TLS in APM Server>>
-. <<https-in-agents,Enable HTTPS in your APM Agents>>
+As soon as secure communication is enabled, requests without a valid token or API key will be denied by APM Server.
+As RUM endpoints cannot be secured, they are exempt from this rule.
+
+* <<api-key,API keys>>
+* <<secret-token,Secret token>>
+
+[[api-key]]
+=== API keys
+
+// Most operations require the `manage_security` cluster privilege,
+// and you must ensure that `apm-server.api_key` or `output.elasticsearch` are configured appropriately.
+// If you modify or delete apm privileges, APM Server might reject all requests.
+// APM Server will create security privileges for the `apm` application; you can freely query them.
+// If you modify or delete apm privileges, APM Server might reject all requests.
+// Check the Elastic Security API documentation for details.
+
+You can configure an API key to authorize requests to the APM Server.
+API keys are sent as plain-text,
+so they only provide security when used in combination with SSL/TLS.
+When you set an API key in each of your agents,
+you ensure that only your agents are able to send data to your APM servers.
```

Comment:
> When you set an API key in each of your agents, you ensure that only your agents are able to send data to your APM servers.

Technically this sentence is not entirely true. How about
> By enabling `apm-server.api_key.enabled: true` you ensure that only agents with a valid API Key are able to successfully use APM Server's API (excempt for RUM endpoints).

Reply:
nice